src/SFA.DAS.ProviderApprenticeshipsService.Application/Domain/Commitment/CommitmentStatusCalculator.cs

```diff
@@ -0,0 +1,119 @@
+﻿using System;
+using SFA.DAS.Commitments.Api.Types;
+using SFA.DAS.Commitments.Api.Types.Commitment.Types;
+using SFA.DAS.ProviderApprenticeshipsService.Application.Exceptions;
+
+namespace SFA.DAS.ProviderApprenticeshipsService.Application.Domain.Commitment
+{
+    public sealed class CommitmentStatusCalculator
+    {
+        /// <remarks>
+        /// Note: any commitments with CommitmentStatus == Active are filtered out before GetStatus is called on them
+        /// </remarks>
+        public RequestStatus GetStatus(EditStatus editStatus, int apprenticeshipCount, LastAction lastAction,
+            AgreementStatus overallAgreementStatus, LastUpdateInfo providerLastUpdateInfo,
+            long? transferSenderId, TransferApprovalStatus? transferApprovalStatus)
+        {
+            if (transferSenderId.HasValue)
+            {
+                if (!transferApprovalStatus.HasValue)
```

Comment:
I think it's best to leave this check in, even if currently there should be no way that we get a valueless transferApprovalStatus when we have a transferSenderId with a value.
The GetStatus call, as well as calculating the status, verifies that the source states are valid and consistent. It might not always be the case that the source states are valid and it's best not to build in assumptions about code outside of this method.